src/cdk-experimental/tree/control/flat-tree-control.ts

```diff
@@ -11,6 +11,24 @@ import {BaseTreeControl} from './base-tree-control';
 /** Flat tree control. Able to expand/collapse a subtree recursively for flattened tree. */
 export class FlatTreeControl<T> extends BaseTreeControl<T> {
 
+  _dataNodes: T[];
+  set dataNodes(newData: T[]) {
+    this._dataNodes = newData;
+
+    let currentParent: T[] = [];
+    for (let i = 0; i < this.dataNodes.length; i++) {
+      const level = this.getLevel(this.dataNodes[i]);
+      currentParent[level] = this.dataNodes[i];
+      if (level > 0) {
+        const parent = currentParent[level - 1];
+        this.parentMap.set(this.dataNodes[i], parent);
+      }
+    }
+  }
+  get dataNodes(): T[] {
```

Comment:
Getter should go before the setter, also with a JsDoc description (dgeni won't pick up the description unless it's on the getter and the getter is first)